app/publish/publish_test.go

```diff
@@ -0,0 +1,125 @@
+	// hook := test.NewLocal(monitor.Logger)
+
+	data := []byte("test file")
+	readSeeker := bytes.NewReader(data)
+	body := &bytes.Buffer{}
+
+	writer := multipart.NewWriter(body)
+
+	fileBody, err := writer.CreateFormFile(fileField, "lbry_auto_test_file")
+	require.Nil(t, err)
+	io.Copy(fileBody, readSeeker)
+
+	jsonPayload, err := writer.CreateFormField(jsonrpcPayloadField)
+	require.Nil(t, err)
+	jsonPayload.Write([]byte(lbrynet.ExampleStreamCreateRequest))
+
+	writer.Close()
+
+	req, err := http.NewRequest("POST", "/", bytes.NewReader(body.Bytes()))
+	require.Nil(t, err)
+
+	req.Header.Set(users.TokenHeader, "uPldrToken")
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	rr := httptest.NewRecorder()
+	authenticator := users.NewAuthenticator(&users.TestUserRetriever{AccountID: "UPldrAcc", Token: "uPldrToken"})
+	publisher := &DummyPublisher{}
+	pubHandler := NewUploadHandler(os.TempDir(), publisher)
+
+	http.HandlerFunc(authenticator.Wrap(pubHandler.Handle)).ServeHTTP(rr, req)
+	response := rr.Result()
+	respBody, _ := ioutil.ReadAll(response.Body)
+
+	assert.Equal(t, http.StatusOK, response.StatusCode)
+	assert.Equal(t, lbrynet.ExampleStreamCreateResponse, string(respBody))
+
+	require.True(t, publisher.called)
+	expectedPath := path.Join(os.TempDir(), "UPldrAcc", ".*_lbry_auto_test_file")
+	assert.Regexp(t, expectedPath, publisher.filePath)
+	assert.Equal(t, "UPldrAcc", publisher.accountID)
+	assert.Equal(t, lbrynet.ExampleStreamCreateRequest, string(publisher.rawQuery))
+	// logEntry := hook.LastEntry()
+	// assert.Equal(t, "lbry_auth_test_file", logEntry.Data["file_name"])
+}
+
+func TestPublishAuthRequired(t *testing.T) {
+	// hook := test.NewLocal(monitor.Logger)
+	var rpcResponse jsonrpc.RPCResponse
+
+	data := []byte("test file")
+	readSeeker := bytes.NewReader(data)
+	body := &bytes.Buffer{}
+
+	writer := multipart.NewWriter(body)
+
+	fileBody, err := writer.CreateFormFile(fileField, "lbry_auto_test_file")
+	require.Nil(t, err)
+	io.Copy(fileBody, readSeeker)
+
+	jsonPayload, err := writer.CreateFormField(jsonrpcPayloadField)
+	require.Nil(t, err)
+	jsonPayload.Write([]byte(lbrynet.ExampleStreamCreateRequest))
+
+	writer.Close()
+
+	req, err := http.NewRequest("POST", "/", bytes.NewReader(body.Bytes()))
+	require.Nil(t, err)
+
+	// req.Header.Set(users.TokenHeader, "uPldrToken")
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	rr := httptest.NewRecorder()
+	authenticator := users.NewAuthenticator(&users.TestUserRetriever{})
+	publisher := &DummyPublisher{}
+	pubHandler := NewUploadHandler(os.TempDir(), publisher)
+
+	http.HandlerFunc(authenticator.Wrap(pubHandler.Handle)).ServeHTTP(rr, req)
+	response := rr.Result()
+
+	assert.Equal(t, http.StatusOK, response.StatusCode)
+	err = json.Unmarshal(rr.Body.Bytes(), &rpcResponse)
+	require.Nil(t, err)
+	assert.Equal(t, "authentication required", rpcResponse.Error.Message)
+	require.False(t, publisher.called)
+	// logEntry := hook.LastEntry()
```

Comment:
I see generally a bunch of commented out code. Most likely from testing but I would remove these in the final product.